android/app/build.gradle

```diff
@@ -94,5 +98,18 @@ flutter {
 }
 
 dependencies {
+    // for flutter_local_notification: to prevent the app from crashing on Android 12L and higher
+    implementation "androidx.window:window:1.0.0"
+    // For Java-friendly APIs to register and unregister callbacks
+    implementation "androidx.window:window-java:1.0.0"
+    // For RxJava2 integration
+    implementation "androidx.window:window-rxjava2:1.0.0"
+    // For RxJava3 integration
+    implementation "androidx.window:window-rxjava3:1.0.0"
+    // For testing
+    implementation "androidx.window:window-testing:1.0.0"
```

Comment:
The [setup instructions](https://pub.dev/packages/flutter_local_notifications) for the flutter_local_notification plugin stated that these are necessary to prevent crashes when desugaring is enabled. It seems to be a flutter issue: [flutter/flutter#110658](https://github.com/flutter/flutter/issues/110658) The lines are copied from  [here](https://developer.android.com/jetpack/androidx/releases/window). I'm not entirely sure if we need them all.